stock_quant_manual_assign/tests/test_stock_quant_manual_assign.py

```diff
@@ -172,3 +172,27 @@ def test_quant_assign_wizard_after_availability_check(self):
             sum(line.qty for line in wizard.quants_lines),
             self.move.reserved_availability,
         )
+
+    def test_quant_assign_unlinks_old_move_line_with_different_lot(self):
+        self.move._action_assign()
+        lot1 = self.env["stock.lot"].create(
+            {
+                "name": "LOT-A",
+                "product_id": self.product.id,
+            }
+        )
+        move_line = self.env["stock.move.line"].create(
+            {
+                "move_id": self.move.id,
+                "product_id": self.product.id,
+                "product_uom_id": self.product.uom_id.id,
+                "qty_done": 1.0,
+                "location_id": self.location1.id,
+                "location_dest_id": self.location_dst.id,
+                "lot_id": lot1.id,
+            }
+        )
+        self.assertIn(move_line, self.move.move_line_ids)
+        wizard = self._create_wizard()
+        move_lines_to_unlink, _ = wizard._get_discrepancies()
+        self.assertEqual(move_lines_to_unlink, move_line)
```

Comment:
This is because there is no stock for `lot1`, correct? In a way, having such a move line record like the one in this test case is unnatural. Would be nice to add some comments to make the intention easier to follow.